strax/storage/s3.py

```diff
@@ -129,27 +133,34 @@ def __init__(self, **kwargs):
         super().__init__()
 
         #  Initialized connection to S3-protocol storage
-        self.s3client = boto3.client(**kwargs,
-                                     service_name='s3')
-        self.kwargs = kwargs
-
-    def get_metadata(self, backend_key):
-        result = self.s3client.get_object(Bucket=backend_key,
-                                          Key='metadata.json')
-
+        self.s3 = boto3.client(**kwargs,
+                               service_name='s3')
+        self.kwargs = kwargs  # Used later for setting up Saver
+
+    def get_metadata(self, key):
+        # Grab metadata object from S3 bucket
+        result = self.s3.get_object(Bucket=BUCKET_NAME,
+                                    Key=f'{key}/metadata.json')
+        # Then read/parse it into ASCII
         text = result["Body"].read().decode()
+
+        # Before returning dictionary
         return json.loads(text)
 
     def _read_chunk(self, backend_key, chunk_info, dtype, compressor):
         with tempfile.SpooledTemporaryFile() as f:
-            self.s3client.download_fileobj(Bucket=backend_key,
-                                           Key=chunk_info['filename'],
-                                           Fileobj=f)
+            self.s3.download_fileobj(Bucket=BUCKET_NAME,
+                                     Key=chunk_info['key_name'],
```

Comment:
OK, so you added a new field `key_name` to the metadata of every chunk, to communicate the strax key to your reach_chunk function (because the storage backend base class first reads the metadata). But you already get the strax key in this function with backend_key -- in your implementation they are exactly the same. So no need to save it again for every chunk.

Reply:
The key name also includes the chunk though.  So key_name is key/chunk

Reply:
OK, so you could do `key = f"{backend_key}/{chunk_info['chunk_i']:06d}"` instead of saving this field in every chunk metadata.